### PR TITLE
feat(ui): add header to tables

### PR DESCRIPTION
**Pull request by UncleGedd**
## Description

Implements Datatable header

## Related Issue

- #84

### CODE REVIEW

File: ui/src/lib/components/DataTable/component.svelte [hunk 1]
```diff
@@ -25,6 +28,24 @@
   const rows = createStore()
   const { namespace, search, searchBy, searchTypes, sortAsc, sortBy } = rows
 
+  // get the title from the route
+  const flatRoutes = routes.flatMap((route) => {
```
Comment (UncleGedd) [inline ui/src/lib/components/DataTable/component.svelte:32]: Nobody look at this....it's super jank and tightly couples the view to the routes (we do this with Breadcrumbs but this feels nasty here)

Going to refactor and pass title + tooltipDesc as props to DataTable

Comment (jeff-mccoy) [reply ui/src/lib/components/DataTable/component.svelte:32]: image

File: ui/src/lib/components/DataTable/component.svelte [hunk 2]
```diff
@@ -33,6 +36,28 @@
...
   <div class="table-container">
     <div class="table-content">
+      <div class="table-header">
+        <h5 class="flex items-center">
+          <span class="dark:text-white">{name}&nbsp;&nbsp;</span>
+          <span class="text-gray-500">
+            {$rows.length} / {$numResources} results
+          </span>
+          <InformationFilled
+            data-tooltip-target="resource-tooltip"
```
Comment (TristanHoladay) [inline ui/src/lib/components/DataTable/component.svelte:46]: not entirely sure, but you might not need these data attributes. in `ui/src/lib/features/k8s/configs/uds-exemptions/exemption-details/component.svelte` there's an example of  tooltip without them.

Comment (jeff-mccoy) [reply ui/src/lib/components/DataTable/component.svelte:46]: Yep those data attributes were for the flowbite js, but our tooltip is pure css if you have the right classes

File: ui/src/lib/utils/descriptions.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2024-Present The UDS Authors
+
+// Generated by robots and the K8s docs
+export const resourceDescriptions: { [key: string]: string } = {
+  ClusterRoleBindings:
```
Comment (TristanHoladay) [inline ui/src/lib/utils/descriptions.ts:6]: this wouldn't work for airgapped, but for connected environments would it be cool to have a link to each resource in the K8s docs?

Comment (UncleGedd) [reply ui/src/lib/utils/descriptions.ts:6]: Good be a nice feature down the line, we'll defer to the air gap use case for now

**Review by Madeline-UX — APPROVED** [review]
I sanction and approve these changes 
- change info icon to not filled
- change number to say "showing x of total"

File: ui/src/lib/components/k8s/DataTable/component.svelte [hunk 4]
```diff
@@ -23,8 +23,17 @@
...
+  let { name, desc } = nameAndDesc
+
   const rows = createStore()
-  const { namespace, search, searchBy, searchTypes, sortAsc, sortBy } = rows
+  const { namespace, search, searchBy, searchTypes, sortAsc, sortBy, numResources } = rows
+
+  // check for filtering
+  let isFiltering = false
+  $: {
+    isFiltering = !!($search || $searchBy != 'Anywhere' || $namespace)
```
Comment (jeff-mccoy) [inline ui/src/lib/components/k8s/DataTable/component.svelte:35]: I wouldn't include searchby as it only has meaning if search has contents

File: ui/src/lib/features/k8s/cluster-ops/hpa/component.svelte [hunk 5]
```diff
@@ -15,6 +17,12 @@
     ['age'],
     ['status'],
   ]
+
+  const resourceName = 'HorizontalPodAutoscalers'
+  const nameAndDesc: NameAndDesc = {
```
Comment (jeff-mccoy) [inline ui/src/lib/features/k8s/cluster-ops/hpa/component.svelte:22]: Placeholder comment to reference in PR review

**Review by jeff-mccoy — CHANGES_REQUESTED** [review]
I left a placeholder comment to see what I'm referring to, but this is every one of those files so bringing up to here instead:

--

This is preference--but feel this would be cleaner as two unique properties and set directly in this file where all presentation parts live for this view, something like 


```
  const name = "HorizontalPodAutoscaler"
  const description = "..."
```

For testing I'm not sure we really need to test every word of the description either vs just the string exists 🤔 but not sure on that. 

Realize that's different than how we did in [kfc](https://github.com/defenseunicorns/kubernetes-fluent-client/blob/main/src/kinds.ts), but think makes more sense in the same file for this project. 

I also wouldn't be object to skipping variable declaration and putting on the component prop directly if it was still readable.

**Comment by UncleGedd** [comment]
> For testing I'm not sure we really need to test every word of the description either vs just the string exists 🤔 but not sure on that.

The goal of the testing  this was to ensure the descriptions matched up to something in `descriptions.ts` (esp since it was generated by 🤖 )

File: ui/src/lib/components/k8s/DataTable/component.svelte (hunk 4 above)
Comment (UncleGedd) [reply ui/src/lib/components/k8s/DataTable/component.svelte:35]: removed

File: ui/src/lib/features/k8s/cluster-ops/hpa/component.svelte (hunk 5 above)
Comment (UncleGedd) [reply ui/src/lib/features/k8s/cluster-ops/hpa/component.svelte]: updated!

**Review by jeff-mccoy — APPROVED** [review]

**Pull request merged**